src/device.cpp

```diff
@@ -363,6 +363,99 @@ void cvk_device::init_features(VkInstance instance) {
     m_features.pNext = pNext;
 }
 
+void cvk_device::init_compiler_options() {
+    m_device_compiler_options = "";
+
+    if (!devices_support_images()) {
+        m_device_compiler_options += " -images=0 ";
+    }
+
+    // 8-bit storage capability restrictions.
+    if (device_8bit_storage_features().storageBuffer8BitAccess == VK_FALSE) {
+        m_device_compiler_options += " -no-8bit-storage=ssbo ";
+    }
+    if (device_8bit_storage_features().uniformAndStorageBuffer8BitAccess ==
+        VK_FALSE) {
+        m_device_compiler_options += " -no-8bit-storage=ubo ";
+    }
+    if (device_8bit_storage_features().storagePushConstant8 == VK_FALSE) {
+        m_device_compiler_options += " -no-8bit-storage=pushconstant ";
+    }
+
+    // 16-bit storage capability restrictions.
+    if (device_16bit_storage_features().storageBuffer16BitAccess == VK_FALSE) {
+        m_device_compiler_options += " -no-16bit-storage=ssbo ";
+    }
+    if (device_16bit_storage_features().uniformAndStorageBuffer16BitAccess ==
+        VK_FALSE) {
+        m_device_compiler_options += " -no-16bit-storage=ubo ";
+    }
+    if (device_16bit_storage_features().storagePushConstant16 == VK_FALSE) {
+        m_device_compiler_options += " -no-16bit-storage=pushconstant ";
+    }
+
+    // Types support
+    if (!supports_fp16()) {
+        m_device_compiler_options += " -fp16=0 ";
+    }
+    if (!supports_fp64()) {
+        m_device_compiler_options += " -fp64=0 ";
+    }
+    if (supports_int8()) {
+        m_device_compiler_options += " -int8 ";
+    }
+    if (supports_ubo_stdlayout()) {
+        m_device_compiler_options += " -std430-ubo-layout ";
+    }
+
+    // Device specific options
+    m_device_compiler_options += m_clvk_properties->get_compile_options();
```

Comment:
`get_compile_options` does not guarantee to return a white-space surrounded string.

Reply:
fixed